src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs

```diff
@@ -1890,7 +1907,11 @@ private bool IsModuleInDenyList(string[] moduleDenyList, string moduleName, Modu
             if (!string.IsNullOrEmpty(moduleName))
             {
                 // moduleName can be just a module name and it also can be a full path to psd1 from which we need to extract the module name
-                exactModuleName = Path.GetFileNameWithoutExtension(moduleName);
+                exactModuleName = moduleName;
+                if (Path.IsPathRooted(moduleName))
```

Comment:
If we use a module name like `./relative/path/to/module.psm1`, will this continue to work with this change? A brief experiment shows that `IsPathRooted` returns false in this case.

Reply:
I found out that we're handling all these cases (1: possible full or relative paths 2: module names with dots in them) in `ModuleIntrinsics.GetModuleName` function, so updated code to using it instead.